src/Data/Collector.ts

```diff
@@ -270,6 +275,87 @@ const isReceiptRobust = async (
   return { success: true }
 }
 
+// Offline receipt verification
+const verifyReceiptOffline = async (
+  receipt: Receipt.ArchiverReceipt, 
+  executionGroupNodes: ConsensusNodeInfo[],
+  minConfirmations: number
+): Promise<{ success: boolean; newReceipt?: Receipt.ArchiverReceipt }> => {
+  const { signedReceipt, globalModification } = receipt
+
+  if (globalModification) {
+    const globalReceipt = signedReceipt as P2PTypes.GlobalAccountsTypes.GlobalTxReceipt
+    // Verify global modification receipt has enough valid signatures
+    const validSigners = new Set<string>()
+    
+    // Ensure signs array exists and is not empty
+    if (!globalReceipt.signs || !Array.isArray(globalReceipt.signs) || globalReceipt.signs.length === 0) {
+      return { success: false }
+    }
+
+    // Ensure tx exists and has required fields
+    if (!globalReceipt.tx || !globalReceipt.tx.address) {
+      return { success: false }
+    }
+
+    for (const sign of globalReceipt.signs) {
+      // Skip invalid signatures
+      if (!sign || !sign.owner) continue
+
+      const node = executionGroupNodes.find(n => n.publicKey === sign.owner)
+      if (!node) continue
+
+      try {
+        if (Crypto.verify({ ...globalReceipt.tx, sign })) {
+          validSigners.add(sign.owner)
+        }
+      } catch (error) {
+        console.error('Error verifying signature:', error)
+        continue
+      }
+    }
+
+    return {
+      success: validSigners.size >= minConfirmations
+    }
+  }
+
+  // Rest of the function remains the same...
+  // Code for normal receipts verification
+  const normalReceipt = signedReceipt as Receipt.SignedReceipt
+  const validSigners = new Set<string>()
+
+  if (!normalReceipt.signaturePack || !Array.isArray(normalReceipt.signaturePack)) {
+    return { success: false }
+  }
+
+  for (const signature of normalReceipt.signaturePack) {
+    if (!signature || !signature.owner) continue
+
+    const node = executionGroupNodes.find(n => n.publicKey === signature.owner)
+    if (!node) continue
+
+    try {
+      const voteHash = calculateVoteHash(normalReceipt.proposal)
+      const appliedVoteHash = {
+        txid: receipt.tx.txId,
+        voteHash
+      }
+
+      if (Crypto.verify({ ...appliedVoteHash, sign: signature })) {
+        validSigners.add(signature.owner)
+      }
+    } catch (error) {
+      console.error('Error verifying signature:', error)
```

Comment:
error message in global tx [validation](https://github.com/shardeum/archive-server/pull/103/files#diff-186c711dbe48d8ffd4c6488e2d29090dae64abd946a5cf185a549af2c556a72eR313) is also the same. we can use differentiator text like `globalTx` & `nonGlobalTx`